# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/tests/test_contenteditable_focus.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 editor/libeditor/tests/test_contenteditable_focus.html
--- a/editor/libeditor/tests/test_contenteditable_focus.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/editor/libeditor/tests/test_contenteditable_focus.html	Thu Sep 22 06:27:37 2022 +0000
@@ -85,10 +85,7 @@
   const otherEditor = document.getElementById("otherEditor");
 
   (function test_initial_state_on_load() {
-    // XXX if there is a contenteditable element, HTML editor sets dom selection
-    // to first editable node, but this makes inconsistency with normal document
-    // behavior.
-    todo_is(
+    is(
       getSelection().rangeCount,
       0,
       "There should be no selection range at start"